packages/jest-snapshot/src/getMatchers.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+const common = require('./common');
+const fs = require('fs');
+const path = require('path');
+const serializer = require('./serializer');
+const Snapshot = require('./Snapshot');
+
+const testsByPath =  {};
+const getNextIndexFor = (test, filePath) => {
+  const tests = testsByPath[filePath] || (testsByPath[filePath] = {});
+  if (test in tests) {
+    return ++tests[test];
+  }
+  return tests[test] = 0;
+};
+
+module.exports = (filePath, options) => ({
+  toMatchSnapshot: (util, customEquality) => {
+    return {
+      compare(actual, expected) {
+        // Retrieving last test asap to avoid race condition
+        const lastTest = common.getLastTest(filePath);
```

Comment:
this whole state management with a single shared global test state seems very brittle to me. What happens if/when we run tests in parallel? I'd much rather hack jasmine to expose this information to custom matchers directly rather than all the monkey patching and global state management. I understand that modifying jasmine isn't super great but the monkey patching doesn't seem so great either.